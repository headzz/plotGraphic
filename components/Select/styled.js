import styled from "styled-components";

export const SelectStyled = styled.select`
  border: none;
  font-family: Lato, Helvetica, sans-serif;
  font-size: 1rem;
  font-weight: 700;
  height: 2rem;
  padding: 4px;
  width: 6rem;

  :focus {
    outline: none;
  }

  div {
    padding: 0.5rem;
  }
`;
