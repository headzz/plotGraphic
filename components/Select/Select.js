import PropTypes from "prop-types";
import React, { useState } from "react";
import { SelectStyled } from "./styled";

/**
 * @component
 * @name Select
 * @description Component that creates a select.
 * @namespace components components/Select
 * @param {object[]} data Data that will create options for the select
 *
 */
const Select = ({ data, handleChange }) => {
  const onChange = (event) => {
    event.preventDefault();
    handleChange(event.target.value);
    window.localStorage.setItem("filterType", event.target.value);
  };

  return (
    <SelectStyled onChange={onChange} data-testid="select">
      {data.map((item) => {
        return (
          <option
            disabled={item.disabled}
            key={item.id}
            defaultValue="0"
            tabIndex={item.id}
            value={item.id}
          >
            {item.name}
          </option>
        );
      })}
    </SelectStyled>
  );
};

export default Select;

Select.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleChange: PropTypes.func.isRequired,
};
