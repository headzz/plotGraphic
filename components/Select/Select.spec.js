import React from "react";
import { render, fireEvent, cleanup, screen } from "@testing-library/react";

import Select from "./Select";
afterEach(cleanup);

const exampleFunction = () => {};

const DATA = [
  { id: 0, name: "1 Ano", disabled: false },
  { id: 1, name: "2 Anos", disabled: false },
];
test("behavior of select content", () => {
  const handleChange = jest.fn((param) => (param += 1));

  const { container, getByTestId } = render(
    <Select handleChange={(event) => handleChange(event)} data={DATA} />
  );

  expect(screen.getByText("1 Ano")).toBeInTheDocument();
  fireEvent.change(screen.getByTestId("select"), {
    target: { value: 1 },
  });

  expect(getByTestId("select")).toHaveTextContent("2 Anos");
  expect(handleChange).toHaveBeenCalledTimes(1);
  expect(container.firstChild).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
});
