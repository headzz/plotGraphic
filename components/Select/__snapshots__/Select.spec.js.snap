// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Select content 1`] = `
<select
  class="sc-bdVaJa cTRebX"
  data-testid="select"
>
  <option
    tabindex="0"
    value="0"
  >
    1 Ano
  </option>
  <option
    tabindex="1"
    value="1"
  >
    2 Anos
  </option>
</select>
`;
